### DISCUSSION TITLE
Form state is lost when the dialog is closed.

### DISCUSSION THREAD

**Discussion by RyanMcErlean** [tailwindlabs/headlessui#893] [Help]
Hello. I am using Dialog with Headless UI Vue. I have a form inside a dialog and I would like to give the user the option to open and close the form.
However, when the dialog is closed the form state is lost.
Normally I would wrap the component in keep-alive (Vue) but that doesn't seem to work. Any ideas?

**Comment by memiljamel** [comment]
have you found a solution for this problem? I experienced it too.

**Comment by LikitSaeLee** [comment]
What works for me is to add `:unmount="false"` on the `<TransitionRoot>`, `<Dialog>`, and `<TransitionChild>` component.

For example
```
 <TransitionRoot as="template" :show="open" :unmount="false">
     ...
     <Dialog :unmount="false">
        ...
        <TransitionChild :unmount="false">
        ...
       </TransitionChild>
      ...
    </Dialog>
   ...
<TransitionRoot>
```

Basically it will prevent the child component from unmounted.
